File: python/47b_894A_QAQ.py
def solve(s):
    cnt = 0
    for i in range(len(s)):
        if s[i] == "Q":
            no_of_A = 0
            for j in range(i+1,len(s)):
                if s[j] == "A":
                    no_of_A += 1
                if s[j] == "Q":
                    cnt += no_of_A
    return cnt

def main():
    n = input()
    print(solve(n))

if __name__ == "__main__":
    main()

File: cpp/47b_894A_QAQ.cpp
#include<bits/stdc++.h>

int solve(std::string s){
    int cnt = 0, NO_of_A;
    for(int i = 0; i < s.size(); i++){
        if (s[i] == 'Q'){
            NO_of_A = 0;
            for(int j = i+1; j < s.size(); j++){
                if (s[j] == 'A'){
                    NO_of_A++;
                }
                if(s[j] == 'Q'){
                    cnt += (NO_of_A);
                }
            }
        }
    }
    return cnt;
}

int main(){
    std::string n;
    std::cin>>n;
    std::cout<<solve(n);
    return 0;
}